android/java/org/chromium/chrome/browser/playlist/kotlin/util/HLSParsingUtil.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.playlist.kotlin.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import androidx.media3.exoplayer.hls.playlist.HlsMediaPlaylist
+import androidx.media3.exoplayer.hls.playlist.HlsMultivariantPlaylist
+import androidx.media3.exoplayer.hls.playlist.HlsPlaylistParser
+
+import java.io.File
+import java.io.FileInputStream
+import java.util.LinkedList
+import java.util.Queue
+
+@SuppressLint("UnsafeOptInUsageError")
+object HLSParsingUtil {
```

Comment:
where do we use that object? I can't find any usage of it.

Reply:
Nvm I found it in `HlsUtils.java`. Do we always use it's methods in the bg thread?

Reply:
we use it mainly for preparing HLS content.

Reply:
there is a case from what I see when we use it from a mojo call, could you check is it on UI thread?

Reply:
Makes sense. i will go through it again.